src/Questar.OneRoster/Filtering/FilterBuilder.cs

```diff
@@ -0,0 +1,137 @@
+namespace Questar.OneRoster.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+
+    public sealed class FilterBuilder<T> : ExpressionVisitor
+    {
+        private readonly Dictionary<ExpressionType, Func<Expression, Expression, FilterBuilder<T>>> _binary;
```

Comment:
It would have to be:
`private static readonly Dictionary<ExpressionType, Func<FilterBuilder<T>, Func<Expression, Expression, FilterBuilder<T>>>> BinaryExpressionToFuncMap;`

Reply:
Moving back to switches and refactoring operators to enums, as discussed.